src/Functions/PolygonsUnion.cpp

```diff
@@ -0,0 +1,126 @@
+#include <Functions/FunctionFactory.h>
+#include <Functions/geometryConverters.h>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+
+#include <common/logger_useful.h>
+
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnTuple.h>
+#include <Columns/ColumnConst.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeCustomGeo.h>
+
+#include <memory>
+#include <string>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int TOO_FEW_ARGUMENTS_FOR_FUNCTION;
+    extern const int BAD_ARGUMENTS;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+
+using CoordinateType = Float64;
+using Point = boost::geometry::model::d2::point_xy<CoordinateType>;
+using Polygon = boost::geometry::model::polygon<Point, false>;
+using MultiPolygon = boost::geometry::model::multi_polygon<Float64Polygon>;
+using Box = boost::geometry::model::box<Point>;
+
+
+class FunctionPolygonsUnion : public IFunction
+{
+public:
+    static inline const char * name = "polygonsUnion";
+
+    explicit FunctionPolygonsUnion() = default;
+
+    static FunctionPtr create(const Context &)
+    {
+        return std::make_shared<FunctionPolygonsUnion>();
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    bool isVariadic() const override
+    {
+        return false;
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 2;
+    }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes &) const override
+    {
+        return DataTypeCustomMultiPolygonSerialization::nestedDataType();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) override
+    {
+        auto get_parser = [&block, &arguments] (size_t i) {
+            const auto * const_col =
+                checkAndGetColumn<ColumnConst>(block.getByPosition(arguments[i]).column.get());
+
+            bool is_const = static_cast<bool>(const_col);
+
+            return std::pair<bool, GeometryFromColumnParser>{is_const, is_const ?
+                makeGeometryFromColumnParser(ColumnWithTypeAndName(const_col->getDataColumnPtr(), block.getByPosition(arguments[i]).type, block.getByPosition(arguments[i]).name)) :
```

Comment:
Don't understand this part -- what is the argument column type, text? And you are parsing it into boost geometry?